RecoLocalCalo/HGCalRecProducers/plugins/HeterogeneousHGCalHEBConditions.cc

```diff
@@ -0,0 +1,177 @@
+#include "RecoLocalCalo/HGCalRecProducers/plugins/HeterogeneousHGCalHEBConditions.h"
+
+HeterogeneousHGCalHEBConditionsWrapper::HeterogeneousHGCalHEBConditionsWrapper(
+    const HGCalParameters* cpuHGCalParameters) {
+  calculate_memory_bytes(cpuHGCalParameters);
+
+  chunk_ = std::accumulate(this->sizes_.begin(), this->sizes_.end(), 0);  //total memory required in bytes
+  cudaCheck(cudaMallocHost(&this->params_.testD_, chunk_));
+
+  //store cumulative sum in bytes and convert it to sizes in units of C++ typesHEB, i.e., number if items to be transferred to GPU
+  std::vector<size_t> cumsum_sizes(this->sizes_.size() + 1, 0);  //starting with zero
+  std::partial_sum(this->sizes_.begin(), this->sizes_.end(), cumsum_sizes.begin() + 1);
+  for (unsigned int i = 1; i < cumsum_sizes.size(); ++i)  //start at second element (the first is zero)
+  {
+    unsigned int typesHEBize = 0;
+    if (cp::typesHEB[i - 1] == cp::HeterogeneousHGCalHEBParametersType::Double)
+      typesHEBize = sizeof(double);
+    else if (cp::typesHEB[i - 1] == cp::HeterogeneousHGCalHEBParametersType::Int32_t)
+      typesHEBize = sizeof(int32_t);
+    else
+      edm::LogError("HeterogeneousHGCalHEBConditionsWrapper") << "Wrong HeterogeneousHGCalParameters type";
```

Comment:
Exception?

Reply:
... or, alternatively, assign a given size in this case, e.g. `sizeof(int32_t);`